File: mobile/lib/dtos/checkout_dto.dart
import 'package:json_annotation/json_annotation.dart';

import 'shipping_rate_dto.dart';

part 'checkout_dto.g.dart';

@JsonSerializable()
class CheckoutDTO {
  String? id;
  double? subtotal;
  double? total;
  ShippingRateDTO? selectedShippingRate;
  List<ShippingRateDTO>? availableShippingRates;

  CheckoutDTO({
    this.id,
    this.subtotal,
    this.total,
    this.selectedShippingRate,
    this.availableShippingRates,
  });

  factory CheckoutDTO.fromJson(Map<String, dynamic> json) =>
      _$CheckoutDTOFromJson(json);

  Map<String, dynamic> toJson() => _$CheckoutDTOToJson(this);
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/api/dto/CheckoutDTO.kt
package me.lucasbatista.vienna.api.dto

data class CheckoutDTO(
    val id: String? = null,
    val subtotal: Double? = null,
    val total: Double? = null,
    val availableShippingRates: List<ShippingRateDTO>? = null,
    val selectedShippingRate: ShippingRateDTO? = null,
)
